octodns/idna.py

```diff
@@ -2,35 +2,84 @@
 #
 #
 
-from idna import decode as _decode, encode as _encode
+from collections.abc import MutableMapping
+
+from idna import IDNAError as _IDNAError, decode as _decode, encode as _encode
 
 # Providers will need to to make calls to these at the appropriate points,
 # generally right before they pass names off to api calls. For an example of
 # usage see https://github.com/octodns/octodns-ns1/pull/20
 
 
+class IdnaError(Exception):
+    def __init__(self, idna_error):
+        super().__init__(str(idna_error))
```

Comment:
This is just used to translate errors thrown by the `idna` lib into something that can be caught inside of octoDNS without other things having to know about that lib.